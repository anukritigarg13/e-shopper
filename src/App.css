.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}
.product-list{
  display:flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin: 5%;
}
.product-type{
  margin: 2%;
}
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
@media screen and (max-width:427px){
  .product-list{
      display:flex;
      justify-content: center;
      flex-wrap: wrap;
  }
}
.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
