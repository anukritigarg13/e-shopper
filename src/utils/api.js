/* eslint-disable import/prefer-default-export */
import axios from 'axios';
import groupByCategory from './helper';

export const getAllItems = async (endpoint) => {
  const { data, error } = await axios.get(endpoint);
  if (data) {
    return groupByCategory(data.data);
  }
  throw new Error(error.message);
};

export const getAllOrders = async (endpoint) => {
  const { data: orders, error } = await axios.get(endpoint);
  if (orders) {
    const categorisedOrders = orders.data.reduce((allOrders, order) => {
      const categorisedOrderItems = groupByCategory(order.items);
      const totalCost = order.items
        .reduce((acc, product) => acc + product.count * product.price, 0);
      const newOrder = {
        ...order, items: categorisedOrderItems, totalItems: order.items.length, totalCost,
      };
      return allOrders.concat(newOrder);
    }, []);
    return categorisedOrders;
  }
  throw new Error(error.message);
};

export const postOrder = async (endpoint, products) => {
  const orderedProducts = Object.keys(products).reduce((accumulator, productCategory) => {
    const productsInCart = products[productCategory]
      .filter((product) => product.itemCount > 0).reduce((acc, product) => {
        const newCount = product.itemCount;
        const newProduct = { ...product, count: newCount };
        delete newProduct.itemCount;
        return acc.concat(newProduct);
      }, []);
    if (accumulator.items === undefined) {
      accumulator.items = [];
    }
    accumulator.items = accumulator.items.concat(productsInCart);
    return accumulator;
  }, {});
  const response = await axios.post('/orders', orderedProducts);
  return response;
};
