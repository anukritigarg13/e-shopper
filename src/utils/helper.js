const groupByCategory = (items) => {
  const categorisedItems = items.reduce((acc, product) => {
    if (acc[product.category] === undefined) {
      acc[product.category] = [];
    }
    const newProduct = { ...product };
    newProduct.itemCount = 0;
    acc[product.category].push(newProduct);
    return acc;
  }, {});
  return categorisedItems;
};

export default groupByCategory;
