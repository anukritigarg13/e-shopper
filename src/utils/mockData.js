/* eslint-disable import/prefer-default-export */
export const mockOrders = {
  data: [
    {
      items: [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 1,
          category: 'Fruits & Vegatables',
        },
      ],
      id: 1,
      date: 1615122360481,
    },
    {
      items: [
        {
          id: 2,
          name: 'table cloth',
          price: 200,
          count: 1,
          category: 'Household Items',
        },
        {
          id: 9,
          name: 'broom',
          price: 250,
          count: 1,
          category: 'Household Items',
        },
      ],
      id: 2,
      date: 1615122649963,
    },
    {
      items: [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 1,
          category: 'Fruits & Vegatables',
        },
        {
          id: 2,
          name: 'table cloth',
          price: 200,
          count: 1,
          category: 'Household Items',
        },
        {
          id: 9,
          name: 'broom',
          price: 250,
          count: 1,
          category: 'Household Items',
        },
        {
          id: 8,
          name: 'curd',
          price: 20,
          count: 2,
          category: 'Dairy & Eggs',
        },
      ],
      id: 3,
      date: 1615122664245,
    },
    {
      items: [
        {
          id: 4,
          name: 'duster',
          price: 80,
          count: 1,
          category: 'Household Items',
        },
        {
          id: 5,
          name: 'milk',
          price: 10,
          count: 2,
          category: 'Dairy & Eggs',
        },
        {
          id: 6,
          name: 'butter',
          price: 20,
          count: 2,
          category: 'Dairy & Eggs',
        },
      ],
      id: 4,
      date: 1615122763596,
    },
    {
      items: [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 3,
          category: 'Fruits & Vegatables',
        },
        {
          id: 3,
          name: 'grapes',
          price: 50,
          count: 3,
          category: 'Fruits & Vegatables',
        },
        {
          id: 7,
          name: 'banana',
          price: 5,
          count: 2,
          category: 'Fruits & Vegatables',
        },
      ],
      id: 5,
      date: 1615540977063,
    },
  ],
};

export const mockProducts = {
  data: [
    {
      id: 1,
      name: 'apple',
      price: 120,
      count: 20,
      category: 'Fruits & Vegatables',
    },
    {
      id: 2,
      name: 'table cloth',
      price: 200,
      count: 3,
      category: 'Household Items',
    },
    {
      id: 3,
      name: 'grapes',
      price: 50,
      count: 10,
      category: 'Fruits & Vegatables',
    },
    {
      id: 4,
      name: 'duster',
      price: 80,
      count: 5,
      category: 'Household Items',
    },
    {
      id: 5,
      name: 'milk',
      price: 10,
      count: 2,
      category: 'Dairy & Eggs',
    },
    {
      id: 6,
      name: 'butter',
      price: 20,
      count: 2,
      category: 'Dairy & Eggs',
    },
    {
      id: 7,
      name: 'banana',
      price: 5,
      count: 48,
      category: 'Fruits & Vegatables',
    },
    {
      id: 8,
      name: 'curd',
      price: 20,
      count: 4,
      category: 'Dairy & Eggs',
    },
    {
      id: 9,
      name: 'broom',
      price: 250,
      count: 4,
      category: 'Household Items',
    },
    {
      id: 10,
      name: 'soap',
      price: 10,
      count: 0,
      category: 'Household Items',
    },
  ],
};

export const categorisedProducts = {
  'Fruits & Vegatables': [
    {
      id: 1,
      name: 'apple',
      price: 120,
      count: 20,
      category: 'Fruits & Vegatables',
      itemCount: 0,
    },
    {
      id: 3,
      name: 'grapes',
      price: 50,
      count: 10,
      category: 'Fruits & Vegatables',
      itemCount: 0,
    },
    {
      id: 7,
      name: 'banana',
      price: 5,
      count: 48,
      category: 'Fruits & Vegatables',
      itemCount: 0,
    },
  ],
  'Household Items': [
    {
      id: 2,
      name: 'table cloth',
      price: 200,
      count: 3,
      category: 'Household Items',
      itemCount: 0,
    },
    {
      id: 4,
      name: 'duster',
      price: 80,
      count: 5,
      category: 'Household Items',
      itemCount: 0,
    },
    {
      id: 9,
      name: 'broom',
      price: 250,
      count: 4,
      category: 'Household Items',
      itemCount: 0,
    },
    {
      id: 10,
      name: 'soap',
      price: 10,
      count: 0,
      category: 'Household Items',
      itemCount: 0,
    },
  ],
  'Dairy & Eggs': [
    {
      id: 5,
      name: 'milk',
      price: 10,
      count: 2,
      category: 'Dairy & Eggs',
      itemCount: 0,
    },
    {
      id: 6,
      name: 'butter',
      price: 20,
      count: 2,
      category: 'Dairy & Eggs',
      itemCount: 0,
    },
    {
      id: 8,
      name: 'curd',
      price: 20,
      count: 4,
      category: 'Dairy & Eggs',
      itemCount: 0,
    },
  ],
};

export const categorisedOrders = [
  {
    items: {
      'Fruits & Vegatables': [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 1,
          category: 'Fruits & Vegatables',
          itemCount: 0,
        },
      ],
    },
    id: 1,
    date: 1615122360481,
    totalItems: 1,
    totalCost: 120,
  },
  {
    items: {
      'Household Items': [
        {
          id: 2,
          name: 'table cloth',
          price: 200,
          count: 1,
          category: 'Household Items',
          itemCount: 0,
        },
        {
          id: 9,
          name: 'broom',
          price: 250,
          count: 1,
          category: 'Household Items',
          itemCount: 0,
        },
      ],
    },
    id: 2,
    date: 1615122649963,
    totalItems: 2,
    totalCost: 450,
  },
  {
    items: {
      'Fruits & Vegatables': [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 1,
          category: 'Fruits & Vegatables',
          itemCount: 0,
        },
      ],
      'Household Items': [
        {
          id: 2,
          name: 'table cloth',
          price: 200,
          count: 1,
          category: 'Household Items',
          itemCount: 0,
        },
        {
          id: 9,
          name: 'broom',
          price: 250,
          count: 1,
          category: 'Household Items',
          itemCount: 0,
        },
      ],
      'Dairy & Eggs': [
        {
          id: 8,
          name: 'curd',
          price: 20,
          count: 2,
          category: 'Dairy & Eggs',
          itemCount: 0,
        },
      ],
    },
    id: 3,
    date: 1615122664245,
    totalItems: 4,
    totalCost: 610,
  },
  {
    items: {
      'Household Items': [
        {
          id: 4,
          name: 'duster',
          price: 80,
          count: 1,
          category: 'Household Items',
          itemCount: 0,
        },
      ],
      'Dairy & Eggs': [
        {
          id: 5,
          name: 'milk',
          price: 10,
          count: 2,
          category: 'Dairy & Eggs',
          itemCount: 0,
        },
        {
          id: 6,
          name: 'butter',
          price: 20,
          count: 2,
          category: 'Dairy & Eggs',
          itemCount: 0,
        },
      ],
    },
    id: 4,
    date: 1615122763596,
    totalItems: 3,
    totalCost: 140,
  },
  {
    items: {
      'Fruits & Vegatables': [
        {
          id: 1,
          name: 'apple',
          price: 120,
          count: 3,
          category: 'Fruits & Vegatables',
          itemCount: 0,
        },
        {
          id: 3,
          name: 'grapes',
          price: 50,
          count: 3,
          category: 'Fruits & Vegatables',
          itemCount: 0,
        },
        {
          id: 7,
          name: 'banana',
          price: 5,
          count: 2,
          category: 'Fruits & Vegatables',
          itemCount: 0,
        },
      ],
    },
    id: 5,
    date: 1615540977063,
    totalItems: 3,
    totalCost: 520,
  },
];
