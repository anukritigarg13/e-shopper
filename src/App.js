import React, { useState, useEffect } from 'react';
import './App.css';
import { Switch, Route } from 'react-router-dom';
import axios from 'axios';
import Navigation from './components/Navigation/Navigation';
// import Cart from './components/Cart/Cart';
// import Checkout from './components/Checkout/Checkout';
// import AllOrders from './components/AllOrders/AllOrders';
//  import { allOrdersData } from './resources/data';
// import ProductList from './components/ProductList/ProductList';
import NavContext from './theme';
import Home from './components/Home/Home';

const App = () => {
  const [products, setProducts] = useState({});
  const [cartItemsCount, setCartItemsCount] = useState(0);
  // const [allOrders] = useState(allOrdersData);
  const [theme, setTheme] = useState('light');
  const [isLoaded, setLoaded] = useState('false');
  const [error, setError] = useState(null);

  useEffect(async () => {
    try {
      const { data, axiosError } = await axios.get('/items');
      if (data) {
        const categorisedProducts = data.data.reduce((acc, product) => {
          if (acc[product.category] === undefined) {
            acc[product.category] = [];
          }
          const newProduct = { ...product };
          newProduct.itemCount = 0;
          acc[product.category].push(newProduct);
          return acc;
        }, {});
        setProducts(categorisedProducts);
        setLoaded(true);
      }
      if (axiosError) {
        setError(axiosError);
        setLoaded(true);
      }
    } catch (e) {
      setError(e);
    }
  }, []);

  const removeItemHandler = (id, category) => {
    const newCategoryProducts = products[category].map((product) => {
      if (product.id === id) {
        if (product.itemCount === 0) {
          return product;
        }
        return {
          ...product,
          itemCount: product.itemCount - 1,
        };
      }

      return product;
    });

    const newProducts = { ...products, [category]: newCategoryProducts };
    setProducts(newProducts);

    const newCartItemsCount = Object.keys(newProducts)
      .reduce((cartTotal, productCategory) => cartTotal + newProducts[productCategory]
        .reduce((cartSubTotal, product) => cartSubTotal + product.itemCount, 0), 0);
    setCartItemsCount(newCartItemsCount);
  };
  const toggleThemeHandler = () => {
    if (theme === 'light') {
      setTheme('dark');
      return;
    }
    setTheme('light');
  };
  const addItemHandler = (id, category) => {
    const newCategoryProducts = products[category].map((product) => {
      if (product.id === id) {
        if (product.itemCount === product.count) {
          return product;
        }
        return {
          ...product,
          itemCount: product.itemCount + 1,
        };
      }
      return product;
    });
    const newProducts = { ...products, [category]: newCategoryProducts };
    setProducts(newProducts);
    const newCartItemsCount = Object.keys(newProducts)
      .reduce((cartTotal, productCategory) => cartTotal + newProducts[productCategory]
        .reduce((cartSubTotal, product) => cartSubTotal + product.itemCount, 0), 0);
    setCartItemsCount(newCartItemsCount);
  };
  if (!isLoaded) {
    return <div>Your data is loading!</div>;
  }
  if (error) {
    return <div>{error}</div>;
  }

  return (
    <div>
      <NavContext.Provider value={theme}>
        <Navigation cartItemsCount={cartItemsCount} toggleTheme={toggleThemeHandler} />
      </NavContext.Provider>

      <Switch>
        <Route exact path="/">
          <Home
            categorisedProducts={products}
            addItemHandler={addItemHandler}
            removeItemHandler={removeItemHandler}
          />
        </Route>
        {/* <Route path="/cart">
          <Cart
            cartItemsCount={cartItemsCount}
            products={products}
          />
        </Route>
        <Route path="/checkout">
          <Checkout />
        </Route>
        <Route path="/allOrders">
          <AllOrders allOrders={allOrders} />
        </Route> */}

      </Switch>

    </div>
  );
};

export default App;
