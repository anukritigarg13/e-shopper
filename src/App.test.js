import {
  fireEvent, render, screen, waitFor, act,
} from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import { getAllOrders, getAllItems } from './utils/api';
import {
  categorisedProducts, categorisedOrders,
} from './utils/mockData';

jest.mock('./utils/api');
describe('App component', () => {
  afterEach(() => {
    jest.clearAllMocks();
    getAllOrders.mockClear();
    getAllItems.mockClear();
  });
  beforeEach(() => {
    getAllOrders.mockResolvedValue(categorisedOrders);
    getAllOrders.mockClear();
    getAllItems.mockResolvedValue(categorisedProducts);
    getAllItems.mockClear();
  });
  it('should match snapshot', async () => {
    const { container } = await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    expect(container).toMatchSnapshot();
  });
  it('should render Home component', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    const homeElement = screen.getAllByTestId('product-list');
    expect(homeElement.length).toBe(3); // 3 product lists: Fruits, Dairy, households
  });
  it('should render Navigation component', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    screen.getByTestId('navigation');
  });
  it('should get products data from backend', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    expect(getAllItems).toHaveBeenCalledTimes(1);
  });
  it('should get orders data from backend', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    expect(getAllOrders).toHaveBeenCalledTimes(1);
  });
  it('should render All Orders page when All Orders clicked', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    const allOrdersElement = screen.getByText('All Orders');
    fireEvent.click(allOrdersElement);
    screen.getByTestId('all-orders');
  });
  it('should render Home page when E-Shopper clicked', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    const eShopperElement = screen.getByText('E-Shopper');
    fireEvent.click(eShopperElement);
    const homeElement = screen.getAllByTestId('product-list');
    expect(homeElement.length).toBe(3);
  });
  it('should render Cart page when My Basket:0 clicked', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    const cartElement = screen.getByText('My Basket :0');
    fireEvent.click(cartElement);
    screen.getAllByTestId('cart');
  });
  it('should increment cart count by 1 when + button clicked', async () => {
    await waitFor(() => render(<BrowserRouter><App /></BrowserRouter>));
    const cartElement = screen.getAllByText('+');
    // console.log(cartElement[0]);
    // screen.logTestingPlaygroundURL();
    act(() => { fireEvent.click(cartElement[0]); });
    screen.getByText('My Basket :1');
  });
});
