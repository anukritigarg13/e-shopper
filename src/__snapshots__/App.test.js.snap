// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div>
  <div>
    <div
      class="navigation"
      data-testid="navigation"
    >
      <div>
        <a
          class="navigation-links"
          href="/"
        >
          E-Shopper
        </a>
      </div>
      <div>
        <button
          type="button"
        >
           Change Theme
        </button>
      </div>
      <div
        class="order-nav"
      >
        <div>
          <a
            class="navigation-links"
            href="/allOrders"
          >
            All Orders
          </a>
        </div>
        <div>
          <a
            class="navigation-links"
            href="/cart"
          >
            My Basket :
            0
          </a>
        </div>
      </div>
    </div>
    <div
      class="product-container"
      data-testid="product-list"
    >
      <div
        class="product-type"
      >
        Fruits & Vegatables
      </div>
      <div
        class="product-list"
      >
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            apple
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              120
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            grapes
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              50
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            banana
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              5
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="product-container"
      data-testid="product-list"
    >
      <div
        class="product-type"
      >
        Household Items
      </div>
      <div
        class="product-list"
      >
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            table cloth
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              200
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            duster
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              80
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            broom
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              250
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="out-of-stock"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            soap
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              10
              /-
            </p>
            <p>
              Out of stock
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="product-container"
      data-testid="product-list"
    >
      <div
        class="product-type"
      >
        Dairy & Eggs
      </div>
      <div
        class="product-list"
      >
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            milk
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              10
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            butter
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              20
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
        <div
          class="product"
          data-testid="product-test"
        >
          <div
            class="product-image-div"
          >
            <img
              alt="image of grapes"
              class="product-image"
              data-testid="product-img"
              src="assets/images/grapes.jpeg"
            />
          </div>
          <p>
            Fresco
          </p>
          <p>
            curd
          </p>
          <p>
            1 kg
          </p>
          <div
            class="quantity-placard"
          >
            <p>
              MRP
               
              20
              /-
            </p>
            <div
              class="quantity-control"
              data-testid="quantity-control"
            >
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  +
                </button>
              </p>
              <p
                class="quantity-in-basket"
              >
                0
                 
                in Basket
              </p>
              <p>
                <button
                  class="control-button"
                  type="button"
                >
                  -
                </button>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
